module/zfs/zfs_znode.c

```diff
@@ -915,7 +915,11 @@ zfs_mknode(znode_t *dzp, vattr_t *vap, dmu_tx_t *tx, cred_t *cr,
 	VERIFY(sa_replace_all_by_template(sa_hdl, sa_attrs, cnt, tx) == 0);
 
 	if (!(flag & IS_ROOT_NODE)) {
-		*zpp = zfs_znode_alloc(zfsvfs, db, 0, obj_type, obj, sa_hdl);
+		do {
+			*zpp = zfs_znode_alloc(zfsvfs, db, 0, obj_type, obj,
```

Comment:
Maybe we could document the fact that we have to do the allocation this way because, even if `kmem_cache_alloc(znode_cache, KM_SLEEP)` should never fail, the VFS can still return a NULL inode if it cannot allocate from the SELinux inode cache in `alloc_inode() -> inode_init_always() -> security_inode_alloc() -> inode_alloc_security()`